scopes/harmony/host-initializer/init-cmd.ts

```diff
@@ -89,11 +95,13 @@ export class InitCmd implements Command {
     if (reset && resetHard) {
       throw new BitError('cannot use both --reset and --reset-hard, please use only one of them');
     }
+
     const workspaceExtensionProps: WorkspaceExtensionProps = {
       defaultDirectory: defaultDirectory ?? getSync(CFG_INIT_DEFAULT_DIRECTORY),
       defaultScope: defaultScope ?? getSync(CFG_INIT_DEFAULT_SCOPE),
-      name,
+      name: name || this.pathToName(path),
```

Comment:
but why so, since this is the place where we pass the workspace props?

Reply:
since we have places where we create workspaces not from the cli but programmatically (like in tests or aspects) this logic should be applied also when using the programmatic API